app/src/main/java/chat/rocket/android/helper/ServerPolicyHelper.java

```diff
@@ -104,9 +92,64 @@ private static boolean isVersionValid(String version) {
     return versionParts.length >= 3 && Integer.parseInt(versionParts[1]) >= 49;
   }
 
+  private static void trySecureValidation(@NonNull OkHttpClient client, @NonNull String host,
+                                          @NonNull Callback callback) {
+    Request request;
+    try {
+      request = createRequest("https://", host);
+    } catch (Exception e) {
+      callback.isNotValid();
+      return;
+    }
+
+    validate(request, client, callback, true);
+  }
+
+  private static void tryInsecureValidation(@NonNull OkHttpClient client, @NonNull String host,
+                                            @NonNull Callback callback) {
+    Request request;
+    try {
+      request = createRequest("http://", host);
+    } catch (Exception e) {
+      callback.isNotValid();
+      return;
+    }
+
+    validate(request, client, callback, false);
+  }
+
+  private static Request createRequest(@NonNull String protocol, @NonNull String host) {
+    return new Request.Builder()
+        .url(protocol + host + API_INFO_PATH)
+        .get()
+        .build();
+  }
+
+  private static void validate(@NonNull Request request, @NonNull OkHttpClient client,
+                               @NonNull Callback callback, boolean usesSecureConnection) {
+    client.newCall(request).enqueue(new okhttp3.Callback() {
+      @Override
+      public void onFailure(Call call, IOException exception) {
+        callback.onNetworkError();
+      }
+
+      @Override
+      public void onResponse(Call call, Response response) throws IOException {
+        if (!response.isSuccessful() || !isValid(response.body())) {
+          callback.isNotValid();
+          return;
+        }
+
+        callback.isValid(usesSecureConnection);
+      }
+    });
+  }
+
   public interface Callback {
-    void isValid();
+    void isValid(boolean usesSecureConnection);
 
     void isNotValid();
+
+    void onNetworkError();
   }
```

Comment:
It would be better to define a class for validation like `ServerPolicyVaridator` because secure/insecure validation has very similar logic.